ruptures/costs/costrank.py

```diff
@@ -95,16 +95,19 @@ def fit(self, signal):
         if signal.ndim == 1:
             signal = signal.reshape(-1, 1)
 
-        # argsort gives us 0-based ranks
+        obs, vars = signal.shape
+
+        # argsort gives us 0-based ranks in the range [0, n-1]. Add one to put it in the range [1, n]
         ranks = np.argsort(signal, axis=0) + 1
```

Comment:
scipy provides a better option than argsort to cope with tied values: `scipy.stats.mstats.rankdata`.
```
from scipy.stats.mstats import rankdata 
...
centered_ranks = rankdata(signal, axis=0) - (obs+1)/2
...
```

Reply:
`scipy.stats.mstats.rankdata` doesn't support any of the tied value options, and `scipy.stats.rankdata` doesn't support per-column ranks, which is why I ended up using argsort.

Reply:
`scipy.stats.mstats.rankdata` implements by default the "average" strategy (the average of the ranks that would have been assigned to all the tied values is assigned to each value) which is more sensible to the "ordinal" strategy (all values are given a distinct rank, corresponding to the order that the values occur in the array) of `np.argsort`.

Reply:
Ah okay, I wasn't sure that averaging the ranks was the behaviour we wanted. Happy to do this.